src/WebUI/Workloads/WorkloadsStore.ts

```diff
@@ -60,17 +61,20 @@ export class WorkloadsStore extends StoreBase {
     }
 
     private _setDeploymentsList = (deploymentsList: V1DeploymentList): void => {
-        this._state.deploymentList = deploymentsList;
-        const deploymentItems = deploymentsList ? deploymentsList.items || [] : [];
-        for (const deployment of deploymentItems) {
-            if (deployment && deployment.metadata.namespace) {
-                this._state.deploymentNamespace = deployment.metadata.namespace;
-                break;
+        const _stateItems = this._state.deploymentList ? this._state.deploymentList.items : [];
+        const _deploymentItems = deploymentsList ? deploymentsList.items : [];
+        const isEquals = Utils.isDeepEquals(_stateItems, _deploymentItems);
+        if (!isEquals || !this._state.deploymentList) {
+            this._state.deploymentList = deploymentsList;
+            const deploymentItems = deploymentsList ? deploymentsList.items || [] : [];
+            for (const deployment of deploymentItems) {
+                if (deployment && deployment.metadata.namespace) {
+                    this._state.deploymentNamespace = deployment.metadata.namespace;
+                    break;
+                }
             }
+            this.emit(WorkloadsEvents.DeploymentsFetchedEvent, this);
         }
-
-        this.emit(WorkloadsEvents.DeploymentsFetchedEvent, this);
-
         if (this._state.deploymentList && this._state.deploymentList.items && this._state.deploymentList.items.length > 0) {
```

Comment:
should not we keep this inside the if condition of "if (!isEquals || !this._state.deploymentList) {"?